src/test/suite/participant/participant.test.ts

```diff
@@ -849,13 +854,27 @@ suite('Participant Controller Test Suite', function () {
                     ],
                   },
                 },
-              ]);
+              ];
+
+              // This is to offset the previous countTokens calls
+              // buildMessages gets called twice for namespace so it is adjusted accordingly
+              // (1 for request prompt and 1 for assistant prompt calculation)
```

Comment:
Not a fan of this offset but `withArgs` gets complicated with the weird VSCode message wrapper that ends up being applied to this point. I tried to use `sinon.match` but seemed to take a lot of time to figure out a way to do this better.

Reply:
Maybe you can list those 4 calls more explicitly in the comment, like:
```
1.
2.
3.
4.
```
to clearly see what those are?